Update metrics docs, add run-container guide (#352)
nginx/documentation: content/agent/how-to/run-agent-container.md

{{< /note >}}

- **Docker:** Ensure Docker is installed and configured on your system. [Download Docker from the official site](https://www.docker.com/products/docker-desktop/).
- **NGINX Agent Image:** You need access to the Docker image for NGINX Agent. Find the appropriate image in your organization's registry or on Docker Hub if publicly available.
- **NGINX Configuration File:** Prepare and validate your NGINX configuration files that the Agent will monitor.
- **Credentials:** Acquire any necessary authentication tokens or credentials required for the NGINX Agent.

1. Pull the NGINX Agent container image

The NGINX Agent container image must be downloaded from a trusted source such as Docker Hub or a private container registry.

Run the following command to pull the official image:

```bash
<!-- Registry HERE -->
docker pull <Registry HERE>:latest
```

Ensure you are using the correct image version. Replace `latest` with the desired version tag if necessary.

2. Create a configuration file

1. Create a configuration file named `nginx-agent.conf` in your current directory.
2. Populate the file with the following structure:

> ```vim

Reviewer: Is this appropriate for a config file? (I don't have a good alternative, ref https://github.com/jincheng9/markdown_supported_languages?tab=readme-ov-file#heres-a-full-list-of-supported-languages)

Reviewer 2: ```suggestion
```yaml
```
yaml works great for this block

command:
  server:
    host: "<NGINX-One-Console-URL>" # Command server host
    port: 443                       # Command server port
  auth:
    token: "<your-data-plane-key-here>" # Authentication token for the command server
  tls:
    skip_verify: false
```

Replace the placeholder values:

- `<NGINX-One-Console-URL>`: The URL of your NGINX One Console instance.
- `<your-data-plane-key-here>`: Your Data Plane access token.

3. Run the container

Run the NGINX Agent container with the configuration file mounted.

Use the following command:

```bash
docker run -d \
  --name nginx-agent \
  -v $(pwd)/nginx-agent.conf:/etc/nginx-agent/nginx-agent.conf \
  nginx/agent:latest
```

Key options explained:

- `-d`: Runs the container in detached mode.
- `--name nginx-agent`: Assigns a name to the container for easy identification.
- `-v $(pwd)/nginx-agent.conf:/etc/nginx-agent/nginx-agent.conf`: Mounts the configuration file into the container.

4. Verify the container is running

Check the running status of the container:

```bash
docker ps
```

> You should see an entry for `nginx-agent`. The `STATUS` field should indicate that the container is running.

Reviewer: ```suggestion
You should see an entry for `nginx-agent`. The `STATUS` field indicates that the container is running.
```

---

> title: "Running the NGINX Agent in a Container"

Reviewer 3: ```suggestion
title: "Run the NGINX Agent in a container"
```

We avoid gerunds, and unnecessary capitalisation: a container is a generic term.

Reviewer 2: ```suggestion
title: "Running NGINX Agent in a container"
```

> weight: 300
> docs: DOCS-000

Reviewer 2: ```suggestion
weight: 300
toc: true
type: how-to
product: Agent
```

---

## Overview

> This guide serves as a step-by-step guide to running the NGINX Agent in a container. It covers the basic setup needed to get the NGINX Agent up and running efficiently and securely.

Reviewer: Does this also apply to the title (line 2)

## Before you begin

Before you begin this guide ensure:

> > **Note**: This guide uses docker but the Agent will also work with other application containers.

Reviewer 3: ```suggestion
{{< note >}}
This guide uses Docker but NGINX Agent will also work with other container applications.
{{< /note >}}
```

- **Installed Docker:** Ensure Docker is installed and configured on your system. [You can download Docker here](https://www.docker.com/products/docker-desktop/).

> - **NGINX Agent Image:** You need access to the Docker image for the NGINX Agent. Find the appropriate image in your organization's registry or on Docker Hub if publicly available.

Reviewer 2: ```suggestion
- **NGINX Agent Image:** You need access to the Docker image for NGINX Agent. Find the appropriate image in your organization's registry or on Docker Hub if publicly available.
```

- **NGINX Configuration File:** Prepare and validate your NGINX configuration files that the Agent will monitor.
- **Credentials:** Acquire any necessary authentication tokens or credentials required for the NGINX Agent.

> 1. Pull the NGINX Agent Container Image

Reviewer 2: ```suggestion
1. Pull the NGINX Agent container image
```

The NGINX Agent container image must be downloaded from a trusted source such as Docker Hub or a private container registry.

Run the following command to pull the official image:

```bash
<!-- Registry HERE -->
docker pull <Registry HERE>:latest
```

Ensure you are using the correct image version. Replace `latest` with the desired version tag if necessary.

> 2. Create a Configuration File

Reviewer 2: ```suggestion
2. Create a configuration file
```
(In general) Only capitalize the 1st word of the sentence, proper nouns, product names, and API objects

1. Create a configuration file named `nginx-agent.conf` in your current directory.
2. Populate the file with the following structure:

```vim
command:
  server:
    host: "<NGINX-One-Console-URL>" # Command server host
    port: 443                       # Command server port
  auth:
    token: "<your-data-plane-key-here>" # Authentication token for the command server
  tls:
    skip_verify: false
```

Replace the placeholder values:

- `<NGINX-One-Console-URL>`: The URL of your NGINX One Console instance.
- `<your-data-plane-key-here>`: Your Data Plane access token.

> 3. Run the Container

Reviewer 2: ```suggestion
3. Run the dontainer
```

Author: Is this correct?

Run the NGINX Agent container with the configuration file mounted.

Use the following command:

```bash
docker run -d \
  --name nginx-agent \
  -v $(pwd)/nginx-agent.conf:/etc/nginx-agent/nginx-agent.conf \
  nginx/agent:latest
```

Key options explained:

- `-d`: Runs the container in detached mode.
- `--name nginx-agent`: Assigns a name to the container for easy identification.
- `-v $(pwd)/nginx-agent.conf:/etc/nginx-agent/nginx-agent.conf`: Mounts the configuration file into the container.

> 4. Verify the Container is Running

Reviewer 2: ```suggestion
4. Verify the container is running
```

Check the running status of the container:

```bash
docker ps
```

You should see an entry for `nginx-agent`. The `STATUS` field should indicate that the container is running.

> 5. Monitor Logs

Reviewer 2: ```suggestion
5. Monitor logs
```
